src/mongodbTypes.ts

```diff
@@ -193,7 +193,7 @@ export type PaprArrayElementsProperties<TSchema> = {
  * }
  */
 export type PaprArrayNestedProperties<TSchema> = {
-  [Property in `${KeysOfAType<PaprAllProperties<TSchema>, Record<string, any>[]>}.$${
+  [Property in `${KeysOfAType<PaprAllProperties<TSchema>, readonly Readonly<Record<string, any>>[]>}.$${
```

Comment:
Why is there two `readonly`s here?

Reply:
I had the same question -- I think `Readonly` makes the record readonly, and then the `readonly` makes the entire array readonly?